### PR TITLE
mutation_compactor: remove emit only live rows parameter

### PR DESCRIPTION
**Pull request by denesb**
Said parameter is a convenience so downstream consumers of the
mutation compactors don't have to check the `bool is_live` already
passed to them. This convenience however causes a template parameter and
additional logic for the compactor. As the most prominent of these
consumers (the query result builder) will soon have to switch to
`emit_only_live_rows::no` for other reasons anyway (it will want to count
tombstones), we take the opportunity to switch everybody to ::no. This
can be done with very little additional complexity to these consumers --
basically an additional if or two. With everybody using the `::no` variant
of the compactor, we can remove this template parameter and the logic
associated with it altogether.

### CODE REVIEW

File: mutation_compactor.hh [hunk 1]
```diff
@@ -293,7 +293,6 @@ public:
         , _collector(std::make_unique<mutation_compactor_garbage_collector>(_schema))
     {
         static_assert(sstable_compaction(), "This constructor can only be used for sstable compaction.");
-        static_assert(!only_live(), "SSTable compaction cannot be run with emit_only_live_rows::yes.");
```
Comment (avikivity) [inline mutation_compactor.hh:306]: You can remove the function only_live() too, and also the comment above the class referring to only_live.

File: querier.cc [hunk 2]
```diff
@@ -287,15 +287,15 @@ void querier_cache::insert_querier(
...
 
-void querier_cache::insert(utils::UUID key, data_querier&& q, tracing::trace_state_ptr trace_state) {
+void querier_cache::insert_data_querier(utils::UUID key, data_querier&& q, tracing::trace_state_ptr trace_state) {
     insert_querier(key, _data_querier_index, _stats, std::move(q), _entry_ttl, std::move(trace_state));
 }
 
-void querier_cache::insert(utils::UUID key, mutation_querier&& q, tracing::trace_state_ptr trace_state) {
+void querier_cache::insert_mutation_querier(utils::UUID key, mutation_querier&& q, tracing::trace_state_ptr trace_state) {
     insert_querier(key, _mutation_querier_index, _stats, std::move(q), _entry_ttl, std::move(trace_state));
 }
```
Comment (avikivity) [inline querier.cc:296]: If data_querier and mutation_querier become the same thing, do we need separate caches for them?

(even if we can merge them, I'd postpone it to a later patch since that's not just a mechanical change)

File: mutation_compactor.hh (hunk 1 above)
Comment (denesb) [reply mutation_compactor.hh:306]: This is done in a later patch. I forgot about the comment though.

File: querier.cc (hunk 2 above)
Comment (denesb) [reply querier.cc:296]: One is used for data queries, the other for mutation queries (used during read-repair). A read-repair will (I think) inherit the UUID of the data query it runs on behalf on, so it is beneficial to separate these into separate indexes, so a cross-lookup becomes impossible. Otherwise we have to rely on one of the many checks to pick up that the lookup found the wrong querier.
With that said, technically, if all the checks pass, the querier created originally for the data read can be used for the mutation read, and the other way around, as there is nothing specific about these anymore.

Comment (avikivity) [reply querier.cc:296]: If there aren't, then we shouldn't have separate caches. It's just complexity.

Note that there won't be reuse in practice, because reconciliation goes back in time. It's possible that the next page of the data query will reuse the mutation query used to reconcile, but it's not very important or likely.

Of course, shouldn't block this series.

Comment (denesb) [reply querier.cc:296]: Note we have a single cache (the reader concurrency semaphore's inactive reader store), the only difference is which index the querier is added to. The index is an intrusive set.
The querier has tree insert/lookup method pairs for data/mutation/shard queriers. Each has its respective index, but the same underlying store.

**Comment by denesb** [comment]
Ping

**Comment by avikivity** [comment]
There was a comment you forgot to update.

File: mutation_compactor.hh (hunk 1 above)
Comment (denesb) [reply mutation_compactor.hh:306]: Done in v2.

**Comment by denesb** [comment]
v2:
* Rebased.
* Removed comment mentioning `emit_only_live_rows` on top of compactor class.
* Removed unused `only_live()` method.
